docs/src/pages/customization/css-in-js.md

```diff
@@ -14,8 +14,8 @@ We think that it's the future:
 - [The future of component-based styling](https://medium.freecodecamp.com/css-in-javascript-the-future-of-component-based-styling-70b161a79a32)
 
 So, you may have noticed in the demos how this *CSS-in-JS* looks.
-We use the [`withStyles`](#api)
-higher-order component to inject an array of styles into the DOM as CSS, using JSS. Here's an example:
+We use the higher-order component created by [`withStyles`](#api)
```

Comment:
`withStyles` _is_ the higher order component.  I don't think this change is correct.

Reply:
Going back to the #9226, it was pointed out that there are 2 conflicting definitions in the docs:

> **Definition 1**
> > We use the `withStyles` higher-order component to inject an array of styles...
>
> **Definition 2**
> > API
> > `withStyles(styles, [options]) => Higher-order Component`

I believe the first definition is correct; it's the second one that needs to change, to say something like
`withStyles(styles, [options]) => Enhanced Component`.

Reply:
I retract my earlier comment, I think this change is correct. `withStyles` is _not_ an HOC, it's simply a function which takes a styles object and _returns_ an HOC, which takes a component and returns a component.

Reply:
@pelotom What about the second half of that sentence? (See the end of my https://github.com/mui-org/material-ui/issues/9226#issuecomment-346187847)

Reply:
@mbrookes you mean this part?

> Would it be more correct to say that the resulting enhanced component injects the CSS? (At what point is the CSS actually injected?)
>
> Since they're all related it would be good pick them up in the associated PR.

I might say something like 

> Calling [`withStyles(styles)`](#api) produces a higher-order component--that is, a function which takes a component and produces a new, enhanced component. The enhancement in this case is to inject class names for the given `styles` into the props of the component it is applied to.